src/test/java/org/dependencytrack/tasks/CloneProjectTaskTest.java

```diff
@@ -0,0 +1,95 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.dependencytrack.tasks;
+
+import org.dependencytrack.PersistenceCapableTest;
+import org.dependencytrack.event.CloneProjectEvent;
+import org.dependencytrack.model.Project;
+import org.dependencytrack.resources.v1.vo.CloneProjectRequest;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.dependencytrack.model.WorkflowStatus.COMPLETED;
+import static org.dependencytrack.model.WorkflowStatus.FAILED;
+import static org.dependencytrack.model.WorkflowStep.PROJECT_CLONE;
+
+public class CloneProjectTaskTest extends PersistenceCapableTest {
+
+    @Test
+    public void testCloneProjectTask() {
+        Project project = qm.createProject("Acme Example", null, "1.0", null, null, null, true, false);
+        CloneProjectRequest request = new CloneProjectRequest(project.getUuid().toString(), "1.1", false, false, false, false, false, false, false, false);
+        final var cloneProjectEvent = new CloneProjectEvent(request);
+        new CloneProjectTask().inform(cloneProjectEvent);
+        var clonedProject = qm.getProject("Acme Example", "1.1");
+        assertThat(clonedProject).isNotNull();
+        assertThat(qm.getAllWorkflowStatesForAToken(cloneProjectEvent.getChainIdentifier())).satisfiesExactly(
+                state -> {
+                    assertThat(state.getStep()).isEqualTo(PROJECT_CLONE);
+                    assertThat(state.getStatus()).isEqualTo(COMPLETED);
+                    assertThat(state.getStartedAt()).isNotNull();
+                    assertThat(state.getUpdatedAt()).isBefore(Date.from(Instant.now()));
+                });
+
+    }
+
+    @Test
+    public void testCloneProjectDoesNotExist() {
+        var uuid = UUID.randomUUID();
+        CloneProjectRequest request = new CloneProjectRequest(uuid.toString(), "1.1", false, false, false, false, false, false, false, false);
+        final var cloneProjectEvent = new CloneProjectEvent(request);
+        assertThatExceptionOfType(IllegalStateException.class)
+                .isThrownBy(() -> new CloneProjectTask().inform(cloneProjectEvent))
+                .withMessageContaining("Project with UUID " + uuid + " was supposed to be cloned, but it does not exist anymore");
```

Comment:
Following my suggestion above to not re-throw the cloning exception, this assertion will no longer work. I'd suggest to instead assert on `state.getFailureReason()`.